imageprocessor/imageprocessor-core/src/main/java/pl/edu/uj/kimage/processingFlow/ImageProcessingFlow.java

```diff
@@ -2,30 +2,47 @@
 
 import pl.edu.uj.kimage.Image;
 import pl.edu.uj.kimage.eventbus.Event;
+import pl.edu.uj.kimage.eventbus.EventBus;
+import pl.edu.uj.kimage.plugin.FlowEvent;
+import pl.edu.uj.kimage.plugin.FlowStep;
+import pl.edu.uj.kimage.plugin.StepResultEvent;
 
 import java.util.Collection;
-import java.util.concurrent.atomic.AtomicInteger;
 
 public class ImageProcessingFlow {
     private final FlowStepRepository flowStepRepository;
-    private AtomicInteger currentStep;
+    private final EventBus eventBus;
+    private final int flowLength;
 
-    //TODO add processing mechanism, event bus
-
-    public ImageProcessingFlow(FlowStepRepository flowStepRepository) {
+    ImageProcessingFlow(FlowStepRepository flowStepRepository, EventBus eventBus, int flowLength) {
         this.flowStepRepository = flowStepRepository;
-        this.currentStep = new AtomicInteger(0);
+        this.eventBus = eventBus;
+        this.flowLength = flowLength;
+    }
+
+    public void start(Image image) {
+        registerFlowToEventBus();
+        StepResultEvent<Image> integerStepResultEvent = new StepResultEvent<>(0, image);
+        eventBus.publish(integerStepResultEvent);
```

Comment:
This is really surprising. `FlowStepRepository` contains `FlowSteps` which contains `EventBus` (I don't know why) and in this class you have another `EventBus`. So you already have two event buses and use only one? 

We talked about two event buses. One for handle processing many (all) images and one for processing each image. Here I see one event bus for each image and second for each `FlowStep`, I don't understand this idea.

Additionally both variables are called `eventBus` which introduces even greater confusion.